### PR TITLE
fix 500 when delete orgnization

### LINKED ISSUE #614: Deleting organisation while user is still in it, causes 500 error
Version: 6aacf4d
OS: Debian stable
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Not relevant

## Description

I've created some organisations, but wanting to keep them 'hidden' from the explore mode to non-logged in users. It seems that this wasn't possible, so I wanted to remove these organisations again.

I had created the orgs with my regular user account, and later removed them with the administrator account.

When logging in with my regular user account, the dashboard, issues and pull-request pages would give 500 errors, explore would work correctly.

It seems that the entries are not removed from the org_user table, after manually removing these, the 500 error disappeared:

sqlite> select * from org_user;
1|1|3|0|0|0
2|2|3|1|1|1
3|2|4|0|1|1
4|2|5|0|1|1

As my only remaining org had org_id 3, I did the following:

sqlite> delete from org_user where org_id > 3;
sqlite> select * from org_user;
1|1|3|0|0|0
2|2|3|1|1|1
sqlite>

this solved this issue, but removing a non-empty org should not be possible, or cascade and remove the relevant org_user entries.

### PR DESCRIPTION
**Pull request by appleboy**
Fix #614 

Error Message

> [...uters/org/setting.go:133 SettingsDelete()] [E] DeleteOrganization: deleteUser: user still has ownership of repositories [uid: 2]

See: https://github.com/go-gitea/gitea/blob/master/models/user.go#L940-L943

ref commit: https://github.com/go-gitea/gitea/commit/ba134bd27abe099efcd748773c410b3aa78521fc

cc @lunny

### CODE REVIEW

**Comment by lunny** [comment]
Why this will resolve the issue?

**Comment by appleboy** [comment]
https://github.com/go-gitea/gitea/blob/master/models/user.go#L940-L943

Must be return error or you will get the following error message.

> [...uters/org/setting.go:133 SettingsDelete()] [E] DeleteOrganization: deleteUser: user still has ownership of repositories [uid: 2]

**Comment by appleboy** [comment]
Reproduce process in https://try.gitea.io/ :

1. Create organization
2. Create a report on the organization
3. Delete organization.

File: models/org.go [hunk 1]
```diff
@@ -233,7 +233,8 @@ func DeleteOrganization(org *User) (err error) {
 	}
 
 	if err = deleteUser(sess, org); err != nil {
-		return fmt.Errorf("deleteUser: %v", err)
+		// Note: don't wrapper error here.
```
Comment (tboerger) [inline models/org.go]: For me it's obvious to not wrapping this error, so IMHO you can drop this comment.

**Comment by tboerger** [comment]
But I still don't get how this will resolve the issue :)

**Comment by lunny** [comment]
I think we need copy function `deleteUser` to `deleteOrg` and remove any no need codes. That's better.

**Comment by appleboy** [comment]
@lunny I will do that.

**Comment by appleboy** [comment]
@lunny Done. I copy function deleteUser to deleteOrg and remove any no need codes.

**Comment by appleboy** [comment]
@tboerger You can see the following code:

https://github.com/go-gitea/gitea/blob/master/routers/org/setting.go#L128-L134

File: models/org.go [hunk 2]
```diff
@@ -224,23 +225,55 @@ func DeleteOrganization(org *User) (err error) {
...
+
+	if err = sess.Commit(); err != nil {
+		return err
+	}
+
+	return RewriteAllPublicKeys()
+}
+
+func deleteOrg(e *xorm.Session, u *User) error {
+	// Check ownership of repository.
```
Comment (lunny) [inline models/org.go]: check u is an org?

File: models/org.go [hunk 3]
```diff
@@ -224,23 +225,55 @@ func DeleteOrganization(org *User) (err error) {
...
-		&TeamUser{OrgID: org.ID},
+	if err = deleteOrg(sess, org); err != nil {
+		return err
+	}
+
+	if err = sess.Commit(); err != nil {
+		return err
+	}
+
+	return RewriteAllPublicKeys()
```
Comment (lunny) [inline models/org.go]: delete an org is no need to RewriteAllPublicKeys() ?

**Comment by lunny** [comment]
So, maybe you can 
```Go
if IsErrUserOwnRepos(err) {
      return err
} else if err != nil {
      return fmt.Errorf("deleteUser: %v", err) 
}
```

**Comment by lunny** [comment]
Otherwise LGTM

**Review by lunny — CHANGES_REQUESTED** [review]

File: models/org.go (hunk 3 above)
Comment (appleboy) [reply models/org.go]: Good catch.

Comment (appleboy) [reply models/org.go]: Done!.

File: models/org.go (hunk 2 above)
Comment (appleboy) [reply models/org.go]: Done.

**Comment by appleboy** [comment]
@lunny Already done.

File: models/org.go (hunk 3 above)
Comment (Bwko) [reply models/org.go]: @lunny [Why did you add it then](https://github.com/appleboy/gitea/commit/ba134bd27abe099efcd748773c410b3aa78521fc#diff-97681c8a2932ea3c23325d3518374d98R239)? :stuck_out_tongue_winking_eye:

Comment (lunny) [reply models/org.go]: 😭 I don't know.

**Review by lunny — APPROVED** [review]

**Comment by Bwko** [comment]
LGTM

**Pull request merged**